src/pymor/algorithms/timestepping.py

```diff
@@ -10,227 +10,619 @@
+by |InstationaryModel|.
 """
 
-from pymor.core.base import ImmutableObject, abstractmethod
+from numbers import Number
+import numpy as np
+
+from pymor.core.base import BasicObject, ImmutableObject, abstractmethod
+from pymor.operators.constructions import IdentityOperator, VectorArrayOperator, ZeroOperator
 from pymor.operators.interface import Operator
+from pymor.parameters.base import Mu
+from pymor.tools import floatcmp
+from pymor.tools.deprecated import Deprecated
 from pymor.vectorarrays.interface import VectorArray
 
 
 class TimeStepper(ImmutableObject):
     """Interface for time-stepping algorithms.
 
-    Algorithms implementing this interface solve time-dependent problems
+    Algorithms implementing this interface solve time-dependent initial value problems
     of the form ::
 
-        M * d_t u + A(u, mu, t) = F(mu, t).
+        M * d_t u + A(u, mu, t) = F(mu, t),
+                     u(mu, t_0) = u_0(mu).
+
+    Time-steppers used by |InstationaryModel| have to fulfill this interface. Time evolution can be performed
+    by calling :meth:`solve`.
 
-    Time-steppers used by |InstationaryModel| have to fulfill
-    this interface.
+    Note that the actual work is done in an iterator derived from :class:`TimeStepperIterator`.
+
+    Parameters
+    ----------
+    num_values
+        The number of returned vectors of the solution trajectory. If `None`, each intermediate vector that is
+        calculated is returned. Else an interpolation of the calculated vectors on an equidistant temporal grid is
+        returned, using an appropriate interpolation of the respective time stepper.
+    interpolation
+        Type of temporal interpolation to be used. Currently implemented are: piecewise constant (P0) and piecewise
+        linear (P1).
     """
 
-    @abstractmethod
-    def solve(self, initial_time, end_time, initial_data, operator, rhs=None, mass=None, mu=None, num_values=None):
+    IteratorType = None
+    available_interpolations = ('P0', 'P1')
+
+    def __init__(self, num_values=None, interpolation='P1'):
+        assert not num_values or (isinstance(num_values, Number) and num_values > 1)
+        assert interpolation in self.available_interpolations
+        self.__auto_init(locals())
+
+    def solve(self, t0, t1, U0, A, F=None, M=None, mu=None, return_iter=False, return_times=False):
         """Apply time-stepper to the equation ::
 
-            M * d_t u + A(u, mu, t) = F(mu, t).
+            M * d_t u + A(u, mu, t) = F(mu, t),
+                         u(mu, t_0) = u_0(mu).
 
         Parameters
         ----------
-        initial_time
+        t0
             The time at which to begin time-stepping.
-        end_time
+        t1
             The time until which to perform time-stepping.
-        initial_data
-            The solution vector at `initial_time`.
-        operator
+        U0
+            The solution vector at `t0`.
+        A
```

Comment:
To match the equation in the init signature. You'd rather have operator here and just reference `A` in the docstring?

Reply:
Yes. (Obviously there are many possible policies for naming such parameters. But I would prefer `operator` over `A´, etc., since I very quickly get confused by all the letters. Thinking of the unification attempts with the system-theoretic models where A,B,C,D,E are so well-established, one might also try to use these letter somehow in a uniform way across pyMOR, but this would certainly be something for another PR, and in most places we have `operator`, `rhs`, etc.)

Reply:
renamed everything back to `operator`, `rhs`, `mass` and `initial_values`